pkg/cmd/install.go

```diff
@@ -594,6 +595,14 @@ func (o *installCmdOptions) setupIntegrationPlatform(c client.Client, namespace
 			}
 		}
 	}
+
+	if platform.Status.Build.BuildConfiguration.ImagePlatforms == nil {
```

Comment:
This must not happen at installation time but at operator runtime. This installation would only cover CLI based installation (which is going to disappear). Beside that, the execution of the `runtime.GOARCH` would be evaluated on the CLI machine, not on the cluster machine. This logic belong to https://github.com/apache/camel-k/blob/main/pkg/platform/defaults.go